### ISSUE TITLE
[Bug] External table Error: Function date_trunc is not implemented when filter time

### ISSUE THREAD

**Issue by gojumprope** [apache/doris#17280]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.2.1

### What's Wrong?

external table only, Execution failed: Error Failed to execute sql: java.sql.SQLException: (conn=1236) errCode = 2, detailMessage = Function date_trunc is not implemented.

### What You Expected?

able to query without error

### How to Reproduce?

```
    SELECT
        date_trunc(localtime, 'hour') __time
    from
		sql_external_table_test
```

__time             |
-------------------+
2022-10-22 07:00:00|

```
SELECT * FROM (
    SELECT
        date_trunc(localtime, 'hour') __time
    from
		sql_external_table_test
) tab
WHERE
	tab.__time >= '2022-10-01 00:00:00'
```

Execution failed: Error Failed to execute sql: java.sql.SQLException: (conn=1236) errCode = 2, detailMessage = Function date_trunc is not implemented.

It can works without the where condition.

```
SELECT * FROM (
    SELECT
        date_trunc(localtime, 'hour') __time
    from
		sql_external_table_test
) tab

```

__time             |
-------------------+
2022-10-22 07:00:00|

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by gojumprope** [comment]
Anyone can look into this issue? It is still not fix in 1.2.3

**Comment by gojumprope** [comment]
Fixed in 2.0.0
